src/MongoDB/Manager.c

```diff
@@ -200,6 +200,15 @@ static void php_phongo_manager_prep_uri_options(zval *options TSRMLS_DC) /* {{{
 				continue;
 			}
 
+            if (!strcasecmp(ZSTR_VAL(string_key), MONGOC_URI_READPREFERENCE)) {
+				ZVAL_DEREF(option);
+				SEPARATE_ZVAL_NOREF(option);
+				if (Z_TYPE_P(option) != IS_STRING) {
+                    phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Unsupported readPreference type, a string must be given");
```

Comment:
> The "cannot be serialized" exception therefore has more to do with ReadPreference not supporting serialize() than being an invalid argument
.......
There is no reason for ReadPreference, WriteConcern, and ReadConcern to not support PHP serialization ....

I see, so basically the problem of the serialization was not with `MongoDB\Driver\ReadPreference` , so something like this `'readpreference' => new MongoDB\Driver\ReadPreference(MongoDB\Driver\ReadPreference::RP_SECONDARY),` should not cause any serialization exceptions.

------------------
so does that would be the proper implementation for this -within `php_phongo_apply_options_to_uri` function ? or should I implement the validation inside `php_phongo_apply_rp_options_to_uri` as this is more related to `readPreference`- ?

```c
if (!strcasecmp(key, MONGOC_URI_READPREFERENCE)) {
    if (!BSON_ITER_HOLDS_UTF8(&iter)) {
        phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "some meaningful exception message");
    }
}
```